engine/src/test/java/org/camunda/bpm/engine/test/history/HistoricProcessInstanceTest.java

```diff
@@ -1924,8 +1973,9 @@ public void testHistoricProcInstQueryWithActivityIdsWithFailingActivity() {
 
     // then
     assertNotNull(result);
-    assertEquals(1, result.size());
-    assertEquals(result.get(0).getId(), failingInstance.getId());
+    assertEquals(2, result.size());
```

Comment:
Now the instance that has not failed yet as in async waiting returned correctly.